cpp/oneapi/dal/algo/knn/common.cpp

```diff
@@ -19,6 +19,12 @@
 #include "oneapi/dal/exceptions.hpp"
 
 namespace oneapi::dal::knn {
+
+namespace optional_results {
```

Comment:
May few algorithms have a shared part of the kernel, like some common primitive in which the flow and result depend on optional_results. I am not sure about ML but in graphs, it can be a traversal case. Should we take some agreement on the order of the options?

Reply:
Assuming these options are each individual for each algorithm.